### PR TITLE
Refine a bit the decentralized merging process.

### PR DESCRIPTION
**Pull request by maximedenes**
We make GitHub assign only principal maintainers as reviewers. This
reduces the level of noise (PRs with 10 code owners), and makes it easy
for the assignee to check if all reviews have been completed (all
reviewers in the list have to approve the PR, which was not the case
before if two reviewers were assigned for the same component).

This change means that when a principal maintainer submits a patch
touching the component they own, they must manually ask a review from the
secondary maintainer.

- [X] Corresponding documentation was added / updated.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (SkySkimmer) [inline .github/CODEOWNERS]: This is incorrect, appveyor.yml is not a dotfile.

Comment (maximedenes) [reply .github/CODEOWNERS]: Ouch! Fixed, thanks a lot.

File: .github/CODEOWNERS
Comment (SkySkimmer) [inline .github/CODEOWNERS]: Why isn't @ejgallego the primary for META.coq?

**Comment by SkySkimmer** [comment]
So secondary maintainership is only for when the primary maintainer is making a PR?

**Comment by maximedenes** [comment]
> So secondary maintainership is only for when the primary maintainer is making a PR?

Yes, or if the primary maintainer wants to delegate the review.

**Comment by maximedenes** [comment]
Note that nothing prevents to permute the maintainers from time to time, if they have constraints.

File: .github/CODEOWNERS
Comment (maximedenes) [reply .github/CODEOWNERS]: Right, seems more logical. Fixed.

**Comment by ejgallego** [comment]
I'm likely misunderstanding, so for example, that means that if I add an extra parameter to a function in `interp`, and adapt all users, then the maintainers of 10 different subsytems have to approve that PR?

**Comment by Zimmi48** [comment]
Theoretically, this is indeed the process @ejgallego but the principal owner (the one who self-assign the PR) may chose to merge without having all approvals in such limit cases.

**Comment by maximedenes** [comment]
In fact, I was surprised that you didn't raise this issue during the meeting. I think we need a way to have an exception for large scale PRs (renamings, propagation of deprecations, etc).

But this is orthogonal to this PR which changes nothing in this respect. So let's discuss it elsewhere.

File: dev/doc/MERGING.md
Comment (Zimmi48) [inline dev/doc/MERGING.md]: You should add a sentence requesting the same thing in case the principal maintainer knows they are not available to do the review.

File: .github/CODEOWNERS
Comment (maximedenes) [reply .github/CODEOWNERS]: Fixed. As for unresponsive maintainers, the file reflects the existing list of maintainers. If it turns out some of them are unresponsive, I suggest we address that separately. I hope to get an an up-to-date list of maintainers as a by-product of this process.

File: dev/doc/MERGING.md
Comment (maximedenes) [reply dev/doc/MERGING.md]: Done, thanks for the suggestion.

**Review by Zimmi48 — APPROVED** [review]

**Comment by maximedenes** [comment]
@ejgallego This PR will make your scenario slightly better (7 owners instead of 14), but I agree we need a better way (an exception for such PRs), so feel free to open an issue.

**Comment by ejgallego** [comment]
> In fact, I was surprised that @ejgallego didn't raise this issue during the meeting (since he's often the author of large scale PRs).

Honestly I am feeling very slow at understanding the changes being proposed lately, I have a hard time "simulating" the new protocol in all the required cases, so I'm reluctant to add a lot of noise until the model becomes clearer in my head. Things happened quite quickly this time.

I'll move discussion back to the main issue, however I am starting to grow concerned about the quite large amount of bureaucratic overhead this new model seems to introduce. I am a fan of very simple things and light methodologies, but if I am understanding correctly, we are gonna need Astérix on our side to get a PR in.

**Comment by maximedenes** [comment]
Vous avez le formulaire bleu ?

**Comment by ejgallego** [comment]
tumblr_mzu4oqy1zn1tpa9smo1_400

**Pull request merged**
